### PR TITLE
Enable use of create_test workflow

### PR DESCRIPTION
**Pull request by gdicker1**
Add definitions of compsets and tests so that typical EarthWorks cases can be run as CIME tests instead.

Testing instructions (in this thread): https://github.com/EarthWorksOrg/EarthWorks/pull/45#issuecomment-2118204061

### CODE REVIEW

**Comment by gdicker1** [comment]
TODOs:
- [ ] Create PRs for changes in externals
    - [ ] https://github.com/gdicker1/mpas-seaice/tree/configdt_highresgrids
    - [ ] https://github.com/gdicker1/ccs_config_cesm/tree/fix_lowres_oQU
- [ ] Add compsets
 	- [ ] "FullyCoupled" -> `2000_CAM60_CLM50%SP_MPASSI_MPASO_SROF_SGLC_SWAV`
	- [ ] "F200climo-mpassi" -> `2000_CAM60_CLM50%SP_MPASSI%PRES_DOCN%DOM_MOSART_CISM2%NOEVOLVE_SWAV`
		- i.e. F2000climo, but use MPAS-seaice prescribed mode instead
	- [ ] "B2000" -> `2000_CAM60_CLM50%SP_MPASSI_MPASO_MOSART_SGLC_SWAV`
- [ ] Expand tests 
    - [x] "smoke test" for FullyCoupled
    - others ...?

**Comment by gdicker1** [comment]
Some info about tests can be found in the CIME Documentation: https://esmci.github.io/cime/versions/master/html/users_guide/testing.html#testname-syntax

File: cime_config/testmods_dirs/earthworks/fullycoupled/user_nl_mpaso [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
+config_cvmix_kpp_use_theory_wave = .true.
```
Comment (gdicker1) [inline cime_config/testmods_dirs/earthworks/fullycoupled/user_nl_mpaso:2]: @dazlich should this variable be set true only under certain conditions (like FullyCoupled at 120km), or do you think this should be true for any EW mpas-o compset? I can create a PR for the second case easily.

**Comment by dazlich** [comment]
I’ve got an outside opinion that it should be generally true.

On May 16, 2024, at 4:57 PM, G. Dylan Dickerson ***@***.***> wrote:


** Caution: EXTERNAL Sender **


@gdicker1 commented on this pull request.

________________________________

In cime_config/testmods_dirs/earthworks/fullycoupled/user_nl_mpaso<https://github.com/EarthWorksOrg/EarthWorks/pull/45#discussion_r1604146390>:

> @@ -0,0 +1,2 @@
+
+config_cvmix_kpp_use_theory_wave = .true.


@dazlich<https://github.com/dazlich> should this variable be set true only under certain conditions (like FullyCoupled at 120km), or do you think this should be true for any EW mpas-o compset? I can create a PR for the second case easily.

**Comment by gdicker1** [comment]
To test these changes:

1. Clone this branch and run `manage_externals`
2. Run `create_test` and point to the `testlist_earthworks.xml` file. Ensure that you point to the EarthWorks input data and supply a suitable (EarthWorks) project/account key. Choose the test category from `ew-pr`, `ew-ver`, and `ew-rel`.
    1. **NOTE:** by default the builds will try to occur on the login node, running the command using the `qcmd` utility on Derecho is recommended to not violate fair use policies.
    2. **NOTE:** `ew-ver` and `ew-rel` can consume a lot of core hours. For now, please add `--no-build` to the `create_test` to ensure these tests only run their setup steps for now
4. Once the cases have been submitted (or finished their appropriate step) you can check the status by going into the directory the tests were created in and running the `cs.status.*` script (the final part of the script name varies). You can `grep` for the "OVERALL" lines or for "FAIL" to limit output.

Example commands to run the `ew-pr` on Derecho and put the tests in `<your_scratch>/ew_pr_tests`
```bash
git clone -b feature/ew-cime-testing https://github.com/gdicker1/EarthWorks
cd EarthWorks
qcmd -A <ew_proj_key> -- ./cime/scripts/create_test--xml-testlist cime_config/testlist_earthworks.xml \
 -i /glade/campaign/univ/ucsu0085/inputdata/ --project <ew_proj_key> \
 --test-root ${SCRATCH}/ew_pr_tests  --xml-category ew-pr
# This will block while builds and other steps finish
cd ${SCRATCH}/ew_pr_tests
./cs.status.*
```

**Review by dazlich — COMMENTED** [review]
@gdicker1 - I've tested this with 
./create_test --xml-testlist ../../cime_config/testlist_earthworks.xml --xml-compiler intel --xml-category ew-pr --project UCSU0085

This creates two cases, FHS94 and FULLYCOUPLED. Both build through compilation. FHS94 also runs, FULLYCOUPLED fails trying to find seaice/ocean data that doesn't exist in the campaign inputdata directory.

**Comment by gdicker1** [comment]
> FULLYCOUPLED fails trying to find seaice/ocean data that doesn't exist in the campaign inputdata directory.

Judging by your command and comment I assume you mean the CESM input data on Campaign? Since I haven't figured out how to add EW as a model within CIME, we will keep getting that (CESM) location by default.

Could you try adding `-i /glade/campaign/univ/ucsu0085/inputdata` and try again?

(Or go into the specific test, run `./xmlchange DIN_LOC_ROOT='/glade/campaign/univ/ucsu0085/inputdata'` and then run `./test.submit`)

**Comment by dazlich** [comment]
@gdicker1 Doh, of course - I am still getting up to speed on 'create_test'

**Comment by dazlich** [comment]
<alias>B2000-EW</alias>

I would have liked to called this CHAOS2000, CHAOS standing for Coupled Hexagonal Atmosphere Ocean and Seaice since that's what the coupled mpas components on a quasi-uniform grid essentially is. ;-)

**Review by dazlich — APPROVED** [review]
Ok, the test I tried before works now adding the --inputdata option.

**Comment by gdicker1** [comment]
> B2000-EW
> 
> I would have liked to called this CHAOS2000, CHAOS standing for Coupled Hexagonal Atmosphere Ocean and Seaice since that's what the coupled mpas components on a quasi-uniform grid essentially is. ;-)

Huh, I thought B2000 was already a compset (could still be, for all I know). I think CHAOS2000 is such a better name! If this is a vote, mine goes to CHAOS2000 instead :D

**Review by dazlich — APPROVED** [review]
Just to add, I ran the entire ew-ver and ew-rel suites with the --no-build option and they all pass.

My intel/fullycoupled ew-pr ran out of time.

**Comment by gdicker1** [comment]
> Just to add, I ran the entire ew-ver and ew-rel suites with the --no-build option and they all pass.
> 
> My intel/fullycoupled ew-pr ran out of time.

@dazlich, thank you. I'll update the FullyCoupled category, it'll only hurt the queue-length for these tests. When you have a sec, could you just make sure the sub-PRs make sense (i.e. pass visual inspection)?

**Comment by dazlich** [comment]
@gdicker1  I've had a look at the sub-PRs and they all look good to me!

**Comment by areanddee** [comment]
I think B2000 is a compset, rings a bell. I vote for CHAOS2000!

**Comment by dazlich** [comment]
Big Boss rejected that name when I proposed it for our CSU model 20 years ago

On May 21, 2024, at 7:21 AM, areanddee ***@***.***> wrote:

I think B2000 is a compset, rings a bell. I vote for CHAOS2000!

**Pull request merged**